platforms/core-configuration/declarative-dsl-core/src/main/kotlin/org/gradle/internal/declarativedsl/parsing/GrammarToTree.kt

```diff
@@ -677,13 +677,7 @@ class GrammarToTree(
                                 }
                             }
 
-                            IDENTIFIER -> {
-                                val receiver = checkForFailure(expression(tree, leftArg!!))
-                                val argument = checkForFailure(valueArgument(tree, rightArg!!))
-                                elementIfNoFailures {
-                                    Element(FunctionCall(checked(receiver), operationTokenName!!, listOf(checked(argument)), tree.sourceData(node)))
-                                }
-                            }
+                            IDENTIFIER -> tree.unsupported(node, UnsupportedLanguageFeature.InfixFunctionCall)
```

Comment:
Would it make sense here to use the `unsupported` overload that accepts two instances of `LighterASTNode`, one for the whole statement, the other for the part that causes the error? 

It would help the tooling maintainers who might want to get a precise pointer to the error location while also having the source data for the whole statement that is invalid.